python/mxnet/contrib/nn/layer.py

```diff
@@ -0,0 +1,294 @@
+# coding: utf-8
+# pylint: disable=
+"""Neural network layers."""
+
+from ... import symbol, ndarray
+from ...symbol import Symbol
+from ...ndarray import NDArray
+from .parameter import Parameter, ParameterDict
+
+
+class Layer(object):
+    """Base class for all neural network layers and models.
+
+    Your models should subclass this class.
+
+    Layers can also contain other Layers, allowing you to nest them in a tree
+    structure. You can assign sublayers as regular attributes::
+        from mxnet import nn
```

Comment:
also suggest to add >>> before each line, then we can have doctest to run all these commands